src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -71,41 +71,79 @@ def get_installed_cli_distributions():
     return [d for d in list(working_set) if d.key == CLI_PACKAGE_NAME or d.key.startswith(COMPONENT_PREFIX)]
 
 
+def _update_latest_from_pypi(versions):
+    from subprocess import check_output, STDOUT, CalledProcessError
+
+    success = False
+    try:
+        cmd = [sys.executable] + '-m pip search azure-cli -vv --disable-pip-version-check --no-cache-dir'.split()
+        logger.debug('Running: %s', cmd)
+        log_output = check_output(cmd, stderr=STDOUT, universal_newlines=True)
+        success = True
+        for line in log_output.splitlines():
+            if not line.startswith(CLI_PACKAGE_NAME):
+                continue
+            comps = line.split()
+            mod = comps[0].replace(COMPONENT_PREFIX, '') or CLI_PACKAGE_NAME
+            version = comps[1].replace('(', '').replace(')', '')
+            try:
+                versions[mod]['pypi'] = version
+            except KeyError:
+                pass
+    except CalledProcessError:
+        pass
+    return versions, success
+
+
 def get_az_version_string():
     import platform
     from azure.cli.core.extension import get_extensions, EXTENSIONS_DIR, DEV_EXTENSION_SOURCES
 
     output = six.StringIO()
-    installed_dists = get_installed_cli_distributions()
+    versions = {}
 
-    cli_info = None
-    for dist in installed_dists:
+    # get locally installed versions
+    for dist in get_installed_cli_distributions():
         if dist.key == CLI_PACKAGE_NAME:
-            cli_info = {'name': dist.key, 'version': dist.version}
-            break
+            versions[CLI_PACKAGE_NAME] = {'local': dist.version}
+        elif dist.key.startswith(COMPONENT_PREFIX):
+            comp_name = dist.key.replace(COMPONENT_PREFIX, '')
+            versions[comp_name] = {'local': dist.version}
+
+    # get the versions from pypi
+    versions, success = _update_latest_from_pypi(versions)
```

Comment:
If `success` is False, why not just return `-1` right here?

Reply:
Because you still need to construct the version string. Returning -1 is just a sentinel to control the final statement. Instead of incorrectly saying "Your CLI is up-to-date" the sentinel will trigger "Unable to determine if your CLI is up-to-date. Check your internet connection."